internal/controller/policymanifest_controller_test.go

```diff
@@ -17,16 +17,197 @@ limitations under the License.
+				Name: kyvernoClusterPolicy.Name,
+			},
+		}
+
+		// Create the Kyverno Cluster Policy and the Giant Swarm Policy Manifest in the cluster
+		Expect(k8sClient.Create(ctx, &kyvernoClusterPolicy)).Should(Succeed())
+		Expect(k8sClient.Create(ctx, &gsPolicyManifest)).Should(Succeed())
+
+		// Initialize the Cluster Policy Reconciler to populate the PolicyCache. Otherwise, the policy manifest reconciliation will fail.
+		_, err := clusterPolicyReconciler.Reconcile(ctx, req)
+		Expect(err).NotTo(HaveOccurred())
+
+	})
+
+	AfterEach(func() {
+		// Clean up the Kyverno Cluster Policy and the Giant Swarm Policy Manifest
+		Expect(k8sClient.Delete(ctx, &kyvernoClusterPolicy)).Should(Succeed())
+		Expect(k8sClient.Delete(ctx, &gsPolicyManifest)).Should(Succeed())
+	})
+
+	Context("When successfully reconciling a PolicyManifest", func() {
+		It("should successfully create a Kyverno Policy Exception", func() {
+			req := ctrl.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      gsPolicyManifest.Name,
+					Namespace: "default",
+				},
+			}
+
+			Expect(policyCache).NotTo(BeEmpty())
+
+			// Test for a successful reconciliation
+			result, err := r.Reconcile(ctx, req)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result.Requeue).To(BeTrue())
 
-		It("should successfully reconcile the resource", func() {
+			// Fetch the Kyverno Policy Exception
+			req = ctrl.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      fmt.Sprintf("gs-kpo-%s-exceptions", gsPolicyManifest.Name),
+					Namespace: "default",
+				},
+			}
+			err = r.Get(ctx, req.NamespacedName, &kyvernoPolicyException)
+			Expect(err).NotTo(HaveOccurred())
 
-			// TODO(user): Add more specific assertions depending on your controller's reconciliation logic.
-			// Example: If you expect a certain status condition after reconciliation, verify it here.
+			// Compare the Kyverno Policy Exception with the expected results
+			Expect(kyvernoPolicyException.Name).To(Equal(fmt.Sprintf("gs-kpo-%s-exceptions", gsPolicyManifest.Name)))
+			Expect(kyvernoPolicyException.Namespace).To(Equal("default"))
+			Expect(kyvernoPolicyException.Labels).To(HaveKeyWithValue("app.kubernetes.io/managed-by", "kyverno-policy-operator"))
+			//Expect(kyvernoPolicyException.Spec.Match.Any[0].Kinds).To(ConsistOf("Deployment", "ReplicaSet", "Pod"))
+			//Expect(kyvernoPolicyException.Spec.Match.Any[1].Kinds).To(ConsistOf("Pod"))
+			Expect(kyvernoPolicyException.Spec.Exceptions[0].PolicyName).To(Equal("disallow-privileged-containers"))
+			Expect(kyvernoPolicyException.Spec.Exceptions[0].RuleNames[0]).To(Equal("restrict-privileged-containers"))
+			Expect(kyvernoPolicyException.Spec.Match.Any[0].ResourceDescription.Names[0]).To(Equal("test-app-1*"))
+			//Expect(kyvernoPolicyException.Spec.Match.Any[0].ResourceDescription.Names[1]).To(Equal("test-app-2*"))
+			Expect(kyvernoPolicyException.Spec.Match.Any[0].ResourceDescription.Namespaces[0]).To(Equal("default"))
```

Comment:
It would be nice to directly compare objects here, but at least we have tests 😄

Reply:
As we discussed, using a follow-up to add a golden file and compare just the specs and some metadata is probably the way to go.